nixos/modules/services/networking/knot.nix

```diff
@@ -98,17 +98,52 @@ in {
         Type = "notify";
         ExecStart = "${cfg.package}/bin/knotd --config=${configFile} --socket=${socketFile} ${concatStringsSep " " cfg.extraArgs}";
         ExecReload = "${knot-cli-wrappers}/bin/knotc reload";
-        CapabilityBoundingSet = "CAP_NET_BIND_SERVICE CAP_SETPCAP";
-        AmbientCapabilities = "CAP_NET_BIND_SERVICE CAP_SETPCAP";
-        NoNewPrivileges = true;
         User = "knot";
+        Group = "knot";
+
+        AmbientCapabilities = [
+          "CAP_NET_BIND_SERVICE"
+        ];
+        CapabilityBoundingSet = [
+          "CAP_NET_BIND_SERVICE"
+        ];
```

Comment:
Dropped `CAP_SETPCAP` since we run unprivileged anyway, and I assume knot does something similar to kresd, for which I don't see the point of carrying this capability.

> CAP_SETPCAP when this capability is available, kresd drops any extra capabilities after the daemon successfully starts when running as a non-root user.

https://knot-resolver.readthedocs.io/en/stable/config-no-systemd-privileges.html

Reply:
With just `CAP_NET_BIND_SERVICE` I agree there's not much point in dropping it after startup.  A different situation is if someone wanted to use the [XDP feature](https://www.knot-dns.cz/docs/3.1/singlehtml/#mode-xdp-pre-requisites), as it requires very powerful capabilities.  But that's just for high-performance deployments; I don't expect it really needs to have an option in the module.

Reply:
Yes, but in that case I read that you will also need several other capabilities, which ultimately could become a knob, if someone is interested. I'm not.

https://www.knot-dns.cz/docs/3.0/singlehtml/#mode-xdp-pre-requisites